bib/cmd/bootc-image-builder/main.go

```diff
@@ -208,6 +221,17 @@ func manifestFromCobra(cmd *cobra.Command, args []string) ([]byte, error) {
 		config = &BuildConfig{}
 	}
 
+	// Disk image types should all share mostly the same manifest but with
+	// different export pipelines.
+	// Right now the qcow2 contains all the pipelines required for ami and raw,
+	// so if one of the image types is qcow2, build that and export pipelines
+	// accordingly if needed.
+	// NOTE: THIS WILL CHANGE WITH THE INTRODUCTION OF NEW IMAGE TYPES
+	imgType := imgTypes[0]
```

Comment:
Could this be simplfied to something like:
```go
diff --git a/bib/cmd/bootc-image-builder/main.go b/bib/cmd/bootc-image-builder/main.go
index 072e132..8b8127e 100644
--- a/bib/cmd/bootc-image-builder/main.go
+++ b/bib/cmd/bootc-image-builder/main.go
@@ -228,7 +228,7 @@ func manifestFromCobra(cmd *cobra.Command, args []string) ([]byte, error) {
        // accordingly if needed.
        // NOTE: THIS WILL CHANGE WITH THE INTRODUCTION OF NEW IMAGE TYPES
        imgType := imgTypes[0]
-       if slices.Contains(imgTypes, "qcow2") {
+       if len(imgTypes) > 1 {
                imgType = "qcow2"
        }
```
(or am I missing something?) - because qcow2 is a superset of all the other images we could always export it. It feels slightly easier to read to me because I don't need to think about what it means if one passes raw,ami.

Reply:
I think I prefer to keep it as is because it makes the logic clear and it might avoid confusion when we need to change it in the future (for example when we bring in other clouds).

> I don't need to think about what it means if one passes raw,ami

I think in this case it's better if we do think about it, because the current code doesn't imply that qcow2 is the superset of all possible disk images.  I mean it is right now, but I'm thinking of the (maybe near-) future scenario when it's not.

On the other hand, we could just always create the qcow2 manifest and simply export what we need.

We should probably eventually make a super-manifest of all the cloud/disk image types and always generate that one.  It would look like this
```mermaid
flowchart
build-->ostree-deployment-->image
image-->qcow2
image-->vmdk
image-->vpc
image-->gcp
image-->hyperv
```
and we'd just match image-type args to exports and be done with it.